### PR TITLE
Minor code updates

### PR DESCRIPTION
**Pull request by BryonLewis**
The biggest thing here is probably replacing the arrow function in Linechart.vue with a named function.  This is to provide the d3.mouse(this) function with the proper 'this' context.  There is another way but it may require messing with babel/webpack settings to be done properly.  The original comment can be found  [HERE](https://github.com/VIAME/VIAME-Web/pull/29#discussion_r408366847)

The only other larger thing was modifying the useEventChart, useMarkerLayer and useTextLayer to have similar checking for the `filteredDetections.value.length` and for them to have similar return values.  All of their components expect to have Arrays.

### CODE REVIEW

File: client/src/use/useTextLayer.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ export default function useTextLayer({
   const vuetify = inject('vuetify');
 
   const textData = computed(() => {
-    if (!filteredDetections.value.length) {
+    if (filteredDetections.value.length === 0) {
       return null;
```
Comment (AlmightyYakob) [inline client/src/use/useTextLayer.js:13]: Is this not supposed to also return `[]`?

Comment (BryonLewis) [reply client/src/use/useTextLayer.js:13]: I would think so, I need to check some of the other files (useTrackFilterControls and others).  I'm assuming we would want to try to maintain returning the same type in the future.  Especially if the component prop is expecting that type.

File: client/src/use/useLineChart.js [hunk 2]
```diff
@@ -4,7 +4,7 @@ export default function useLineChart({ filteredDetections, typeColorMap }) {
   const lineChartData = computed(() => {
     const _filteredDetections = filteredDetections.value;
     if (_filteredDetections.length === 0) {
-      return null;
+      return [];
```
Comment (subdavis) [inline client/src/use/useLineChart.js:7]: You changed this from a falsey to a truthy value, causing the template `v-if` to change meaning.

File: client/src/use/useMarkerLayer.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ export default function useMarkerLayer({ filteredDetections, selectedTrackId })
 
   const markerData = computed(() => {
     if (filteredDetections.value.length === 0) {
-      return null;
+      return [];
```
Comment (subdavis) [inline client/src/use/useMarkerLayer.js:9]: You changed this from a falsey to a truthy value, causing the template `v-if` to change meaning.

File: client/src/use/useTrackFilterControls.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ export default function useFilteredTracks({ detections }) {
 
   const filteredDetections = computed(() => {
     if (!detections.value.length) {
-      return null;
+      return [];
```
Comment (subdavis) [inline client/src/use/useTrackFilterControls.js:11]: Agree with this change.

File: client/src/use/useTextLayer.js [hunk 5]
```diff
@@ -9,8 +9,8 @@ export default function useTextLayer({
   const vuetify = inject('vuetify');
 
   const textData = computed(() => {
-    if (!filteredDetections.value.length) {
-      return null;
+    if (filteredDetections.value.length === 0) {
+      return [];
```
Comment (subdavis) [inline client/src/use/useTextLayer.js:13]: You changed this from a falsey to a truthy value, causing the template `v-if` to change meaning.

**Pull request merged**